lib/rom/sql/extensions/postgres/types/range.rb

```diff
@@ -0,0 +1,56 @@
+require 'sequel/core'
+
+Sequel.extension(:pg_range)
+
+module ROM
+  module SQL
+    module Postgres
+      module Values
+        Range = ::Struct.new(:lower, :upper, :lower_bound, :upper_bound)
+      end
+
+      module Types
+        def self.Range(name, subtype)
+          Type(name) do
+            read = SQL::Types.Constructor(Values::Range) do |value|
+              pg_range =
+                if value.is_a?(Sequel::Postgres::PGRange)
+                  value
+                elsif value.respond_to?(:to_s)
+                  Sequel::Postgres::PGRange::Parser
```

Comment:
since parsers are like pure functions we should cache them somewhere, a global cache would be fine